packages/jest-parallel/src/index.js

```diff
@@ -0,0 +1,182 @@
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+'use strict';
+
+const mergeStream = require('merge-stream');
+const path = require('path');
+
+const JestWorker = require('./worker');
+
+import type {JestApi, JestFarmOptions} from './types';
+import type {Readable} from 'stream';
+
+/**
+ * The Jest farm is a class that allows you to queue methods across multiple
+ * child processes, in order to parallelize work. This is done by providing an
+ * absolute path to a module that will be loaded on each of the child
+ * processes, and bridged to the main process.
+ *
+ * Bridged methods are specified by using the "exposedMethods" property of the
+ * options "object". This is an array of strings, where each of them corresponds
+ * to the exported name in the loaded module.
+ *
+ * You can also control the amount of workers by using the "workers" property of
+ * the "options" object, and the settings passed to fork the process through
+ * the "forkOptions" property.
+ *
+ * Queueing calls can be done in two ways:
+ *   - Standard method: calls will be redirected to the first available worker,
+ *     so they will get executed as soon as they can.
+ *
+ *   - Sticky method: if a "getWorkerKey" method is provided within the config,
+ *     the resulting string of this method will be used as a key. Everytime this
+ *     key is returned, it is guaranteed that your job will be processed by the
+ *     same worker. This is specially useful if your workers are caching
+ *     results.
+ */
+class JestFarm {
+  _aggregagedStdout: Readable;
+  _aggregagedStderr: Readable;
+  _api: JestApi;
+  _ending: boolean;
+  _hashes: {[string]: JestWorker};
+  _options: JestFarmOptions;
+  _workers: Array<JestWorker>;
+
+  constructor(options: JestFarmOptions) {
+    const api = this._makeCall.bind(this, '');
+    const aggregatedStdout = mergeStream();
+    const aggregatedStderr = mergeStream();
+    const workers = [];
+
+    if (!path.isAbsolute(options.workerPath)) {
+      throw new TypeError('Worker path needs to be absolute');
+    }
+
+    // Build the options once for all workers to avoid allocating extra objects.
+    const workerOptions = {
+      forkOptions: options.forkOptions,
+      workerPath: options.workerPath,
+    };
+
+    for (let i = 0; i < options.workers; i++) {
+      const jestWorker = new JestWorker(workerOptions);
+
+      aggregatedStdout.add(jestWorker.getStdout());
+      aggregatedStderr.add(jestWorker.getStderr());
+
+      workers.push(jestWorker);
+    }
+
+    options.exposedMethods.forEach(name => {
+      api[name] = this._makeCall.bind(this, name);
+    });
+
+    this._aggregagedStdout = aggregatedStdout;
+    this._aggregagedStderr = aggregatedStderr;
+    this._api = api;
+    this._ending = false;
+    this._hashes = Object.create(null);
+    this._options = options;
+    this._workers = workers;
+  }
+
+  getApi(): JestApi {
+    return this._api;
+  }
+
+  getAggregatedStdout(): Readable {
```

Comment:
I think just `getStdout` is sufficient as a name for this, no? It's implied that it is the aggregated data.